### PR TITLE
FP-270: Sticky Header Component

### PR DESCRIPTION
**Pull request by robert-arias**
# Ticket(s)

- [FP-270: Product - FED](https://fourkitchens.clickup.com/t/36718269/FP-270)

## Purpose
- Create a sticky header

### Functional Testing
- [x] See: https://github.com/fourkitchens/forcepoint-nextjs-page-router/pull/35

### CODE REVIEW

File: src/lib/components/02-components/sticky-header/sticky-header.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { ElementType } from "react";
+import { cn } from "../../../utils/tailwind-merge";
+import Button, { ButtonColorVariant } from "../../01-elements/button/button";
+import Link from "../../01-elements/link/link";
+import Typography from "../../01-elements/typography/typography";
+import Branding from "../branding/branding";
+
+const stickyHeaderLogo =
+  'https://www.forcepoint.com/sites/all/themes/custom/fp/assets/img/logos/f-white.svg';
```
Comment (mcortes19) [inline src/lib/components/02-components/sticky-header/sticky-header.tsx:9]: We definitely need this svg as a local asset

File: src/lib/components/02-components/sticky-header/sticky-header.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+export default function StickyHeader({
+  headerTitle,
+  headerCtas,
+  showHeader = true,
+  linkComponent = 'a',
+  headerLogoAlt,
+}: StickyHeaderProps) {
+  const ctas = headerCtas.map((cta, index) => {
```
Comment (mcortes19) [inline src/lib/components/02-components/sticky-header/sticky-header.tsx:32]: could you please rename this variable as `renderedCtas`?

File: docs/api/stickyHeader.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+The StickyHeader component interface allows you to create sticky headers within the user interface.
+
+## Import
+
+```js
+import { StickyHeader } from 'forcepoint-shared-components';
+```
+
+export type StickyHeaderProps = {
```
Comment (mcortes19) [inline docs/api/stickyHeader.md:11]: This section doesn't have code formatter, and not sure if we need it at all

File: docs/api/stickyHeader.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  headerTitle?: string;
+  headerCtas: HeaderCta[];
+  showHeader: boolean;
+  linkComponent?: ElementType;
+  headerLogoAlt: string;
+};
+
+export type HeaderCta = {
```
Comment (mcortes19) [inline docs/api/stickyHeader.md:19]: This section doesn't have code formatter, and not sure if we need it at all

**Review by mcortes19 — COMMENTED** [review]
@robert-arias great job! I just left some comments.

I also noticed there are some code formatting changes when I saved files. So heres's [patch](https://github.com/fourkitchens/forcepoint-shared-components/files/14716035/code-format.patch). Please make sure your VSCode is configured to avoid this type of conflicts

File: docs/api/stickyHeader.md (hunk 4 above)
Comment (robert-arias) [reply docs/api/stickyHeader.md:19]: Yes, I left this by mistake, I was using it to create the table. Sorry! And good catch.

**Review by mcortes19 — APPROVED** [review]

**Pull request merged**